crates/ironrdp-svc/src/lib.rs

```diff
@@ -126,81 +133,132 @@ pub trait StaticVirtualChannel: AsAny + fmt::Debug + Send + Sync {
 
 assert_obj_safe!(StaticVirtualChannel);
 
-/// Takes a vector of PDUs and breaks them into chunks prefixed with a Channel PDU Header (`CHANNEL_PDU_HEADER`).
-///
-/// Each chunk is at most `max_chunk_len` bytes long (not including the Channel PDU Header).
-pub fn chunkify(messages: Vec<SvcMessage>, max_chunk_len: usize) -> PduResult<Vec<WriteBuf>> {
-    let mut results = vec![];
-    for message in messages {
-        results.extend(chunkify_one(message, max_chunk_len)?);
-    }
-    Ok(results)
+/// SvcPreprocessor is used to chunkify/de-chunkify static virtual channel PDUs.
+#[derive(Debug)]
+pub struct SvcPreprocessor {
+    /// Buffer for de-chunkification of clipboard PDUs. Everything bigger than ~1600 bytes is
+    /// usually chunked when transfered over svc.
+    chunked_pdu: Option<Vec<u8>>,
```

Comment:
Tell me if I’m wrong: `Option` is used so that it’s easy to `take` the inner vector out and to hand it to the user, but at no point you really need to handle a `None` state (a new vector is always inserted when `None`). This sounds like you could use `Vec<u8>` directly, and `std::mem::take` it instead. This will leave a default vector in place (no allocation is performed for empty vectors, created using `Vec::new`, `Vec::default` or `vec![]`), and remove the need to `unwrap` though I’m confident there was no risk of panicking in this case.

Reply:
5409cbf6302508d1d384b0065545f4904479db0c